raulin0/school-manager
# Feature/api

## Description
This pull request introduces a series of changes to enhance the project's documentation, configuration, and functionality related to a student registration system. It also includes the addition of third-party packages for improved API documentation and introduces CRUD operations for student, course, and registration databases.

## Changes Made
- **Documentation Improvements**:
  - Added comprehensive installation and usage instructions to the README.md, making it easier for users to set up and utilize the project.
  - Included project settings updates and URL enhancements in project documentation.
  - Integrated settings with Swagger API documentation for better API documentation.
  - Added information about the GNU General Public License version 3 (GPL-3.0) to the documentation.

- **Functionality Enhancements**:
  - Added key packages like `rest_framework`, `drf_yasg`, `api`, and `django_filters` to `INSTALLED_APPS` in project settings, extending the project's capabilities.
  - Introduced endpoints in the `urls.py` file to access various views.
  - Created views to support CRUD (Create, Read, Update, Delete) operations for the student, course, and registration databases.
  - Implemented serializers and input validators to enhance the student database's functionality.
  - Defined the structure and relationships for the student, course, and registration databases.

- **Code Quality and Style**:
  - Improved code readability by replacing double quotes with single quotes in `manage.py`.

## Motivation and Context
These changes aim to improve the project in several ways:
- **User-Friendliness**: The addition of detailed installation and usage instructions in the README makes it easier for users to get started.
- **Documentation**: The project's documentation is updated to reflect the latest changes and third-party package integrations.
- **API Enhancement**: The inclusion of `rest_framework`, `drf_yasg`, and other packages improves API capabilities and documentation.
- **Functionality**: CRUD operations for key databases enhance the project's core functionality.
- **Licensing**: Properly documented the project's licensing information.

## How to Test
1. Review the updated README for installation and usage instructions.
2. Check the project's settings and URLs for any updates.
3. Confirm that Swagger API documentation now reflects the project's settings.
4. Verify that the newly added packages (`rest_framework`, `drf_yasg`, `django_filters`, etc.) are correctly integrated into `INSTALLED_APPS`.
5. Test the endpoints introduced in `urls.py` for accessing views.
6. Perform CRUD operations on the student, course, and registration databases using the newly created views.
7. Ensure that input validation and serialization work as expected in the student database.
8. Confirm that the database structures and relationships are correctly defined.
9. Review the updated documentation for licensing details.
10. Check for code style improvements by ensuring that double quotes are replaced with single quotes in `manage.py`.

## Related Issue(s)
N/A
